rest_framework/routers.py

```diff
@@ -235,36 +251,44 @@ def get_urls(self):
 
-            for route in routes:
-
-                # Only actions which actually exist on the viewset will be bound
-                mapping = self.get_method_map(viewset, route.mapping)
-                if not mapping:
-                    continue
-
-                # Build the url pattern
-                regex = route.url.format(
-                    prefix=prefix,
-                    lookup=lookup,
-                    trailing_slash=self.trailing_slash
-                )
-
-                # If there is no prefix, the first part of the url is probably
-                #   controlled by project's urls.py and the router is in an app,
-                #   so a slash in the beginning will (A) cause Django to give
-                #   warnings and (B) generate URLS that will require using '//'.
-                if not prefix and regex[:2] == '^/':
-                    regex = '^' + regex[2:]
+            is_viewset = isclass(viewset) and issubclass(viewset, ViewSetMixin)
 
-                initkwargs = route.initkwargs.copy()
-                initkwargs.update({
-                    'basename': basename,
-                    'detail': route.detail,
-                })
+            for route in routes:
 
-                view = viewset.as_view(mapping, **initkwargs)
-                name = route.name.format(basename=basename)
-                ret.append(re_path(regex, view, name=name))
+                if is_viewset:
+                    # Only actions which actually exist on the viewset will be bound
+                    mapping = self.get_method_map(viewset, route.mapping)
+                    if not mapping:
+                        continue
+
+                    # Build the url pattern
+                    regex = route.url.format(
+                        prefix=prefix,
+                        lookup=lookup,
+                        trailing_slash=self.trailing_slash
+                    )
+
+                    # If there is no prefix, the first part of the url is probably
+                    #   controlled by project's urls.py and the router is in an app,
+                    #   so a slash in the beginning will (A) cause Django to give
+                    #   warnings and (B) generate URLS that will require using '//'.
+                    if not prefix and regex[:2] == '^/':
+                        regex = '^' + regex[2:]
+
+                    initkwargs = route.initkwargs.copy()
+                    initkwargs.update({
+                        'basename': basename,
+                        'detail': route.detail,
+                    })
+
+                    view = viewset.as_view(mapping, **initkwargs)
+                    name = route.name.format(basename=basename)
+                    django_path = re_path(regex, view, name=name)
+                else:
+                    # assume that `viewset` is a Django view
+                    django_path = path(prefix, viewset, name=prefix)
```

Comment:
note to self: move this out of the for-loop? I think currently `ret` is getting a bunch of repeat `path`s when iterating over non-DRF views